### DISCUSSION TITLE
help needed: persisting a field using local storage

### DISCUSSION THREAD

**Discussion by HFolkertH** [cincheo/daquota#28] [Q&A] [answered]
hi all

I am doing a simple test. Entering a input-field and hopefully save it to Local-storage. Next time i open the app, that saved value should be in input-field. But it doesn't. What am i ding wrong?
{
  "layout": "block",
  "components": [
    {
      "key": "test1",
      "defaultValue": "=[]",
      "type": "LocalStorageConnector",
      "eventHandlers": [],
      "cid": "local-storage-0"
    },
    {
      "label": "",
      "inputType": "text",
      "description": "",
      "field": "Veld",
      "size": "default",
      "disabled": false,
      "placeholder": "",
      "type": "InputView",
      "eventHandlers": [
        {
          "global": false,
          "name": "@click",
          "actions": [
            {
              "targetId": "$self",
              "name": "eval",
              "description": "Default action",
              "argument": "target.dataModel.veld = $d('input-2')"
            }
          ]
        }
      ],
      "cid": "input-2",
      "dataSource": "local-storage-0",
      "defaultValue": "=''"
    }
  ],
  "defaultValue": "={}",
  "type": "ContainerView",
  "eventHandlers": [],
  "cid": "index"
}

**Comment by renaudpawlak** [comment, marked as answer]
Local storage are not made (yet?) to contain simple values such as string or integers. They have to contain objects or arrays.
For your use case, bind the container's source to the local storage and the input to its parent container. Also, initialize the local storage with an empty object ({}) as a default value. 

IMPORTANT NOTE: since you local storage may be in an inconsistant state, you might have to go to the data tab of the local storage and press the reset button to make it right.

Find the modified container below:

```json
{
  "layout": "block",
  "components": [
    {
      "key": "test1",
      "defaultValue": "={}",
      "type": "LocalStorageConnector",
      "eventHandlers": [],
      "cid": "local-storage-0"
    },
    {
      "label": "",
      "inputType": "text",
      "description": "",
      "field": "Veld",
      "size": "default",
      "disabled": false,
      "placeholder": "",
      "type": "InputView",
      "eventHandlers": [
        {
          "global": false,
          "name": "@click",
          "actions": [
            {
              "targetId": "$self",
              "name": "eval",
              "description": "Default action",
              "argument": "target.dataModel.veld = $d('input-0')"
            }
          ]
        }
      ],
      "cid": "input-0",
      "dataSource": "$parent",
      "defaultValue": "=''"
    }
  ],
  "defaultValue": "={}",
  "type": "ContainerView",
  "eventHandlers": [],
  "cid": "container-0",
  "dataSource": "local-storage-0"
}
```

**Reply by HFolkertH** [reply]
Hello Renaud,

Thanks for helping me out. Your app works as I hoped. Good and simple start study material ;-)
You use the @Click action on the input field.  I did nit know that @Click also reacts on Enter and Tab key. That's why I was experimenting with @Update, @Change.  Is there any documentation about these @actions?

Another thing: I noticed the default value for the Local-storage key is now "{}".  But I also see "[]". I guess the first one is for single objects and the latter one for Object Collections?  Perhaps an explanation text with that attribute?
